Feat/port 74.5.35

pabloarodas opened the pull request:
* Add validation to corsPolicy's allowed URIs when creating/updating identity zones #179727364
* fix: flyway mysql migration SQL script failure when using utf8mb4 character set #181291079

hsinn0 committed:
Add validation to corsPolicy's allowed URIs when creating/updating identity zones

- Currently, the `CorsFilter` only validates the global corsPolicy's allowed URIs
at server start time (https://github.com/pivotal/lts-uaa/blob/811ab79ec99742fbdd216154a29b626180b2be61/server/src/main/java/org/cloudfoundry/identity/uaa/security/web/CorsFilter.java#L101) and disregards the per-zone cors policy.
We are about to adapt `CorsFilter` to respect the per-zone cors policy.
And since per-zone cors policy is added/edited at server's run time,
it is problematic/inefficient to validate the per-zone cors policy as
the requests come through `CorsFilter`; hence we are validating the cors
policy when the zone is created/edited.
- Currently, the `CorsFilter` validates the URIs by compiling them into
regex patterns. We are keeping this validation approach (even though we
are not sure why the currrent code is treating these allowed URIs values
as regex patterns, especially when there are separate fields called
`allowedUriPatterns` and `allowedOriginPatterns`).

[#179727364]

Co-authored-by: Peter Chen <peterch@vmware.com>

Commit:
Refactor: remove unused cors filter in sample app

- we are about to adapt this cors filter for #179727364 which will
impact its usage in the sample app; but then we find no evidence
that the cors filter is needed in the sample app, so we opt to remove
it.
- the cors filter is specifically written for the UAA server; if the
sample app needs a cors policy filter (we don't think it needs it),
it should have its own logic.

[#179727364]

Co-authored-by: Hongchol Sinn <hsinn@vmware.com>

hsinn0 committed:
fix: flyway mysql migration SQL script failure when using utf8mb4 character set

- In mysql8, the default charset is utf8mb4, which is max 4 bytes. This caused a flyway migration script SQL statement to fail because it makes the row size go over the limit.
- Removed the SQL statement that failed because it is overwritten by later migration script anyhow
- Modified the database test script to create uaa mysql databases with defalut character set utf8mb4 so the too large raw size scenario can be tested
- Modified the database start script to honor db parameter value of 'mysql-5'

Co-authored-by: Peter Chen <peterch@vmware.com>
Co-authored-by: Danny Faught <dfaught@vmware.com>

[#181291079]

strehle commented on server/src/main/java/org/cloudfoundry/identity/uaa/zone/GeneralIdentityZoneConfigurationValidator.java:
```diff
@@ -83,6 +91,18 @@ public IdentityZoneConfiguration validate(IdentityZone zone, IdentityZoneValidat
+        if (uris != null) {
+            for (String uri : uris) {
+                try {
+                    Pattern.compile(uri);
```
sonar suggests that you do Pattern.compile(Pattern.quote(uri));

https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&types=VULNERABILITY&pullRequest=1966&id=cloudfoundry-identity-parent&open=AYI7Se-6BHqZnT1RNIiH

pabloarodas replied:
We tried doing that change but tests started failing after that. Since this is complaining about a value that is not a user input, what we did was ignore the specific validation for this method.

swalchemist replied:
Yeah, the tests probably require that these strings actually work with regex metacharacters, which is what Sonar is complaining about.

pabloarodas committed:
fix: ignore non applicable sonar validation javasecurity:S2631

- The value sonar is complaining about is not something the user can
  input or change

Co-authored-by: Pablo Rodas <prodas@vmware.com>
Co-authored-by: Ramkumar Vengadakrishnan <rvengadakrishnan@pivotal.io>
